drivers/reset/reset_mmio.c

```diff
@@ -0,0 +1,100 @@
+/* Copyright (c) 2025 Google LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/reset.h>
+
+struct reset_mmio_dev_config {
+	uint32_t base;
+	uint8_t num_resets;
+	bool active_low;
+};
+
+static inline int reset_mmio_status(const struct device *dev, uint32_t id, uint8_t *status)
+{
+	const struct reset_mmio_dev_config *config = dev->config;
+	uint32_t value;
+
+	if (id >= config->num_resets) {
+		return -EINVAL;
+	}
+
+	value = sys_read32(config->base);
+	*status = FIELD_GET(BIT(id), value);
+
+	/* If active low, invert the logic */
+	if (config->active_low) {
+		*status = !*status;
```

Comment:
```suggestion
		*status = !（*status）;
```

This seems much more relaxed？

Reply:
Done